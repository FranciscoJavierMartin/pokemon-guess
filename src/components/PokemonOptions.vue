<template>
  <div class="options-container">
    <ul>
      <li
        v-for="option in options"
        :key="option.id"
        @click="choosePokemon(option.id)"
      >
        {{ option.name }}
      </li>
    </ul>
  </div>
</template>

<script lang="ts">
import { Pokemon } from '@/interfaces/pokemon';
import { defineComponent } from 'vue';

export default defineComponent({
  props: {
    options: {
      type: Array as () => Pokemon[],
    },
  },
  emits: ['selection'],
  methods: {
    choosePokemon(id: number): void {
      this.$emit('selection', id);
    },
  },
});
</script>

<style scoped>
ul {
  list-style-type: none;
}
li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>
