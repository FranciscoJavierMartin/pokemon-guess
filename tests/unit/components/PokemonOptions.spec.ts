import { shallowMount, VueWrapper } from '@vue/test-utils';
import PokemonOptions from '@/components/PokemonOptions.vue';
import { pokemons } from '../mocks/pokemons.mock';

describe('PokemonOptions', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, { props: { options: pokemons } });
  });

  test('Snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should four options', () => {
    const liTags = wrapper.findAll('li');
    expect(liTags).toHaveLength(4);
    pokemons.forEach((pokemon, index) => {
      expect(liTags[index].text()).toBe(pokemon.name);
    });
  });

  test('should emit "selection" event when click on one option', () => {
    const liTags = wrapper.findAll('li');

    liTags.forEach((li) => {
      li.trigger('click');
    });
    expect(wrapper.emitted('selection')).toHaveLength(pokemons.length);
    pokemons.forEach((pokemon, index) => {
      expect(wrapper.emitted('selection')![index]).toEqual([pokemon.id]);
    });
  });
});
