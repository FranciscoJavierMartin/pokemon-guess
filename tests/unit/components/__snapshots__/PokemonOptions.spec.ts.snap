// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>pikachu</li>
    <li>bulbasur</li>
    <li>charmander</li>
    <li>squirtle</li>
  </ul>
</div>
`;
